File: CRUD/VB.Net/Aplicacion con SQL Server V2/Sistema/Sistema/Forms/FormCambiarUsuario.vb
' Written By Ismael Heredia in the year 2017

Imports Sistema.Aplicacion.Data

Public Class FormCambiarUsuario

    Dim usuarioDatos As New UsuarioDatos
    Dim funciones As New Funciones
    Dim usuario As String = ""

    Public Sub New(ByVal usuario_recibido As String)
        InitializeComponent()
        usuario = usuario_recibido
    End Sub

    Private Sub btnCambiarUsuario_Click(sender As Object, e As EventArgs) Handles btnCambiarUsuario.Click
        If txtUsuarioActual.Text = "" OrElse txtClave.Text = "" OrElse txtNuevoNombre.Text = "" Then
            MessageBox.Show("Faltan datos")
        Else
            Dim usuario As String = txtUsuarioActual.Text
            Dim clave As String = funciones.md5_encode(txtClave.Text)
            Dim nuevo_usuario As String = txtNuevoNombre.Text

            Dim id As Integer = usuarioDatos.get_id_by_user(usuario)

            If usuarioDatos.check_login(usuario, clave) Then

                Dim grabar_ready As Boolean = False

                If usuarioDatos.check_exists_usuario_add(nuevo_usuario) Then
                    grabar_ready = False
                Else
                    grabar_ready = True
                End If

                If grabar_ready Then
                    If usuarioDatos.change_username(id, nuevo_usuario) Then
                        MessageBox.Show("El nombre de usuario ha sido cambiado , reinicie la aplicación")
                        Application.Exit()
                    Else
                        MessageBox.Show("Ha ocurrido un error en la base de datos")
                    End If
                Else
                    MessageBox.Show("El usuario " & nuevo_usuario & " ya existe")
                End If
            Else
                MessageBox.Show("Login inválido")
            End If
        End If
    End Sub


    Private Sub FormCambiarUsuario_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        txtUsuarioActual.Text = usuario
    End Sub

End Class

File: CRUD/C#/Aplicacion con SQL Server V2/Aplicacion/Forms/FormCambiarUsuario.cs
// Written By Ismael Heredia in the year 2017

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Aplicacion.Data;

namespace aplicacion
{
    public partial class FormCambiarUsuario : Form
    {

        UsuarioDatos usuarioDatos = new UsuarioDatos();
        Funciones funcion = new Funciones();
        String usuario = "";

        public FormCambiarUsuario(String usuario_recibido)
        {
            InitializeComponent();
            usuario = usuario_recibido;
        }

        private void btnCambiar_Click(object sender, EventArgs e)
        {
            if (txtUsuarioActual.Text == "" || txtClave.Text == "" || txtNuevoNombre.Text == "")
            {
                MessageBox.Show("Faltan datos");
            }
            else
            {
                string usuario = txtUsuarioActual.Text;
                string clave = funcion.md5_encode(txtClave.Text);
                string nuevo_nombre = txtNuevoNombre.Text;

                int id = usuarioDatos.get_id_by_user(usuario);

                if (usuarioDatos.check_login(usuario,clave))
                {

                    bool grabar_ready = false;

                    if (usuarioDatos.check_exists_usuario_add(nuevo_nombre))
                    {
                        grabar_ready = false;
                    }
                    else
                    {
                        grabar_ready = true;
                    }

                    if (grabar_ready)
                    {
                        if (usuarioDatos.change_username(id,nuevo_nombre))
                        {
                            MessageBox.Show("El nombre de usuario ha sido cambiado , reinicie la aplicación");
                            Application.Exit();
                        }
                        else
                        {
                            MessageBox.Show("Ha ocurrido un error en la base de datos");
                        }
                    }
                    else
                    {
                        MessageBox.Show("El usuario " + nuevo_nombre + " ya existe");
                    }
                }
                else
                {
                    MessageBox.Show("Login inválido");
                }
            }
        }

        private void FormCambiarUsuario_Load(object sender, EventArgs e)
        {
            txtUsuarioActual.Text = usuario;
        }
    }
}
